src/fu-common.c

```diff
@@ -993,3 +994,396 @@ fu_common_get_path (FuPathKind path_kind)
 
 	return NULL;
 }
+
+#define AS_UTILS_DECODE_BCD(val)	((((val) >> 4) & 0x0f) * 10 + ((val) & 0x0f))
+
+/**
+ * as_utils_version_from_uint32:
+ * @val: A uint32le version number
+ * @flags: flags used for formatting, e.g. %AS_VERSION_PARSE_FLAG_USE_TRIPLET
+ *
+ * Returns a dotted decimal version string from a 32 bit number.
+ *
+ * Returns: A version number, e.g. "1.0.3"
+ *
+ * Since: 0.5.2
+ **/
+gchar *
+as_utils_version_from_uint32 (guint32 val, AsVersionParseFlag flags)
```

Comment:
What were you thinking to do with stuff like `AsVersionParseFlags`?  I suspect you'll want to have enums come to fwupd for this type of stuff.